File: examples/01_strip_on.py
# import modules:
import board
import neopixel # neopixel.py should be in /lib

# create pixels object:
pixels = neopixel.NeoPixel(
    board.GP0,
    9,
    brightness=0.3,
    auto_write=False
)

# set color for the entire strip (red):
pixels.fill((255, 0, 0))

# update the physical led strip:
pixels.show()

# try other colors:
RED = (255, 0, 0)
GREEN = (0, 255, 0)
BLUE = (0, 0, 255)
YELLOW = (255, 255, 0)
CYAN = (0, 255, 255)
MAGENTA = (255, 0, 255)

File: arduino/01_strip_on.cpp
#include <FastLED.h> // FastLED library should be installed

#define PIN 6            // pin where the to which the data line of the LED strip is connected
#define NUM_PIXELS 10    // number of LEDs on the strip
CRGB pixels[NUM_PIXELS]; // our pixels array used to control the strip

void setup()
{
    // initalize the LED strip
    FastLED.addLeds<NEOPIXEL, PIN>(pixels, NUM_PIXELS);

    // set brightness (a bit lower than max to save our eyes)
    FastLED.setBrightness(100); // (min: 0, max: 255)

    // clear previous data from the strip
    FastLED.clearData();

    CRGB colorPredefined = CRGB::Red;     // uses predefined colors
    CRGB colorCustom = CRGB(255, 0, 255); // uses custom RGB value

    // set color for the entire strip
    fill_solid(pixels, NUM_PIXELS, colorPredefined);

    // update the physical led strip
    FastLED.show();
}

void loop()
{
    // do nothing yet
}